ERCS/erc-7596.md

```diff
@@ -0,0 +1,2756 @@
+        uint classOfShare  
+    ) external view returns(uint32);
+
+    function headOfList(
+        uint classOfShare
+    ) external view returns (uint32);
+
+    function tailOfList(
+        uint classOfShare
+    ) external view returns (uint32);
+
+    function lengthOfList(
+        uint classOfShare
+    ) external view returns (uint);
+
+    function getSeqList(
+        uint classOfShare
+    ) external view returns (uint[] memory);
+
+    function getChain(
+        uint classOfShare
+    ) external view returns (GoldChain.NodeWrap[] memory);
+
+    // ==== Order ====
+
+    function isOrder(
+        uint classOfShare,
+        uint seqOfOrder
+    ) external view returns (bool);
+    
+    function getOrder(
+        uint classOfShare,
+        uint seqOfOrder
+    ) external view returns (GoldChain.Node memory );
+
+    // ==== Class ====
+
+    function isClass(uint classOfShare) external view returns(bool);
+
+    function getClassesList() external view returns(uint[] memory);
+```
+
+## Rationale
+
+The System is designed based on legal principles of corporate laws and commercial laws that are generally accepted by major jurisdictions:
+1. Constitutional documents (i.e. Shareholders Agreement in the System) **SHALL** be approved by founding members of the company concerned, and update of which **SHALL** be subject to approval of members in accordance with the voting rules stipulated by itself;
+2. All legal and commercial rights of shareholders (or members), conditions and procedures of exercising the shareholders' rights, hiearachy structure of the company as well as corporate governance rules concerned, all those aspects of a company **SHALL** be set out and governed by provisions of the Constitutional documents (i.e. Shareholders Agreement);
+3. Register of Shares privately managed by the company shall be the fundamental and decisive records to create and verify the legal identity and economic rights of shareholders;
+4. Shareholders (or members), as wealth owner of the equity shares, **SHALL** have rights to transfer, pledge, create options or other legal encumbrances upon the equity shares they owned;
+5. Shareholders (or members), as beneficial owner of the company, **SHALL** have rights to engage in corporate governance activites, including but not limited to proposing motions, voting on motions, nominating candidates (of directors or managers), convening meetings, executing motions etc.;
+6. Agreements **SHALL** have binding forces upon the signing parties; and
+7. other legal and commercial principles that can hardly be exaustively specified.
+
+## Backwards Compatibility
+
+This is the first edition of this ERC governing the on-chain behaviors of members and stakeholders of a company with respect to equity share transaction and corporate governance.  However, the later updated versions **SHALL** keep compatible to the current edition in views of legal, commercial and technical consistance.
```

Comment:
You can't really create requirements on future proposals. People can (and will) do whatever they want.